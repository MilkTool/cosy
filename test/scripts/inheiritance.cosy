
class Weapon {
    describe() {
        print "This can be used to hurt people with."
    }
}

class Axe < Weapon {}

var axe = Axe()
axe.describe()

print "--------"

class Sword < Weapon {
    describe() {
        super.describe()
        print "Poke them with the pointy end."
    }
}

var sword = Sword()
sword.describe()
