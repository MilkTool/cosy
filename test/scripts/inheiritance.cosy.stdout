This can be used to hurt people with.
--------
This can be used to hurt people with.
Poke them with the pointy end.
